### ISSUE TITLE
Release jquery-rails 4.0

### ISSUE THREAD

**Issue by chancancode** [rails/jquery-rails#168]
**TL;DR** We need to release a new version of `jquery-rails` soon™ to support Rails 4.2. Apps running Rails 4.1 and below can continue to use the 3.x releases.
## Reason

The main reason for this release is to fix #167. Since this is a breaking change (the implementation changed from `HTML::Document` to `Nokogiri`, which we expose via `@selected`, we need to bump the major version.

So on `jquery-rails` 4.0+ we will implement `assert_select_jquery` via nokogiri, and we will keep the `HTML::Document` version on the 3.x releases.

Since we don't have to (and can't) support Rails 4.1 and below with this new version, allegedly this would also allow us to clean up some of the code as well.
## Da Plan
- [x] Create 3-1-stable branch
- [x] Bump master to 4.0.0
- [x] Require railties ">= 4.2.0.beta" (for now - we need to release a new version that dep 4.2.0 later because bundler)
- [x] Fix #167. @rubys has [a patch](https://github.com/rubys/jquery-rails/commit/94298a39bd77bd31d867cc7544ea9edf3996647d) ready for this.
- [x] :scissors: Rails 3 stuff from master (@JangoSteve @lucasmazza)
- [x] Release `jquery-rails` 4.0.0
- [x] Make Rails 4.2 applications generate the Gemfile with `gem 'jquery-rails', '>= 4.0.0'` (or `~> 4.0`?)
